src/centralserver/admin-service/src/test/java/org/niis/xroad/centralserver/restapi/openapi/SubsystemsApiTest.java

```diff
@@ -86,6 +91,46 @@ public void addSubsystem() {
         }
     }
 
+    @Nested
+    @DisplayName("GET " + "/api/v1/members/{id}/subsystems")
+    @WithMockUser(authorities = {"VIEW_MEMBER_DETAILS"})
+    class GetSubsystems {
+
+        @Test
+        public void getSubsystems() throws JsonProcessingException {
+            final String subsystemCode = UUID.randomUUID().toString();
+            ClientIdDto subsystemIdDto = (ClientIdDto) new ClientIdDto()
+                    .memberClass("GOV")
+                    .memberCode("M2")
+                    .subsystemCode(subsystemCode)
+                    .instanceId("TEST")
+                    .type(XRoadIdDto.TypeEnum.SUBSYSTEM);
+            ClientDto subsystemDto = new ClientDto()
+                    .xroadId(subsystemIdDto);
+
+            var createResponse = restTemplate.postForEntity(PATH, subsystemDto, ClientDto.class);
+            assertEquals(HttpStatus.CREATED, createResponse.getStatusCode(), "Failed to create susbsystem");
+
+            var response = restTemplate.getForEntity(
+                    "/api/v1/members/M2/subsystems",
+                    String.class);
```

Comment:
I think you can provide the type here, instead of String. There would be no need for manual mapping to json later.